scripts/ci/run_perftest.sh

```diff
@@ -117,36 +157,78 @@ for scale in 50 100 500; do
     # run influxql and flux query tests
     for query_file in $query_files; do
       format=$(echo $query_file | cut -d '-' -f7)
+      format=${format%.txt}
       cat ${DATASET_DIR}/$query_file | ${GOPATH}/bin/query_benchmarker_influxdb --urls=http://localhost:8086 --debug=0 --print-interval=0 --workers=$workers --json=true --organization=$TEST_ORG --token=$TEST_TOKEN | jq ". += {branch: \"${INFLUXDB_VERSION}\", commit: \"${TEST_COMMIT}\", time: \"$datestring\", i_type: \"${DATA_I_TYPE}\", query_format: \"$format\"}" > $working_dir/test-query-$scale_string-format-$format-workers-$workers.json
     done
 
-    # delete db to start anew
-    delete_response=$(curl -s -o /dev/null -X DELETE -H "Authorization: Token $TEST_TOKEN" "http://${NGINX_HOST}:8086/api/v2/buckets/$bucket_id?org=$TEST_ORG" -w %{http_code})
-    if [ $delete_response != "204" ]; then
-      echo "bucket not deleted!"
-      exit 1
-    fi
+    clean_db
   done
 done
 
+# Metaquery tests
+
+# Controls how many data points will be generated. The total number of points
+# will be scale_var * scale_var, where each point is (approximately) a unique
+# series.
+scale_var=1000
+
+# How many queries to generate.
+queries=100
+
+# The time range controls both the span over which data is generated, and the
+# span over which queries will be performed. timestamp-start and timestamp-end
+# must be provided to both the data generation and query generation commands
+# and must be the same to ensure that the queries cover the data range.
+start="2019-01-01T00:00:00Z"
+end="2020-01-01T00:00:00Z"
+
+# How long to run the query phase of the benchmarks. It's best to specify a
+# duration to limit the maximum amount of time the queries can run for these,
+# since individual queries can take a long time to run.
+duration=30s
+
+# Generate data
+scale_string="scalevar-$scale_var"
+scale_seed_string="$scale_string-seed-$seed"
+data_fname="influx-bulk-records-usecase-metaquery-$scale_seed_string.txt"
+$GOPATH/bin/bulk_data_gen --seed=$seed -use-case metaquery -scale-var $scale_var -timestamp-start $start -timestamp-end $end > ${DATASET_DIR}/$data_fname
+
+# Generate flux queries for field-keys and tag-values
+query_files=""
+for type in field-keys tag-values; do
+  for format in http flux-http; do
+    query_fname="query-metaquery-$scale_seed_string-$type-$format.txt"
+    $GOPATH/bin/bulk_query_gen -use-case metaquery -query-type $type -format influx-$format -timestamp-start $start -timestamp-end $end -queries $queries > ${DATASET_DIR}/$query_fname
+    query_files="$query_files $query_fname"
+  done
+done
+
+# Load the data
+load_opts="-urls=http://${NGINX_HOST}:8086 -do-abort-on-exist=false -do-db-create=true"
+if [ -z $INFLUXDB2 ] || [ $INFLUXDB2 = true ]; then
+  load_opts="$load_opts -organization=$TEST_ORG -token=$TEST_TOKEN"
+fi
+cat ${DATASET_DIR}/$data_fname | $GOPATH/bin/bulk_load_influx $load_opts > /dev/null
+
+# compacting is skipped for now on metaquery tests, which have many shards
```

Comment:
Skipping compaction here for now; need to figure out why it's not working for this test. I think it might have something to do with the many shards that the data creates. I'll circle back around to this soon but want to get this update merged so that other tests can build off of it.